lib/auth/userpreferences/userpreferencesv1/service.go

```diff
@@ -95,3 +97,25 @@ func (a *Service) UpsertUserPreferences(ctx context.Context, req *userpreference
 
 	return &emptypb.Empty{}, trace.Wrap(a.backend.UpsertUserPreferences(ctx, username, req.Preferences))
 }
+
+// GetKeyboardLayout returns the keyboard layout preference for the user.
+func (a *Service) GetKeyboardLayout(ctx context.Context, req *userpreferences.GetKeyboardLayoutRequest) (*userpreferences.GetKeyboardLayoutResponse, error) {
+	authCtx, err := a.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	identity := authCtx.Identity.GetIdentity()
+
+	if !slices.Contains(identity.SystemRoles, string(types.RoleWindowsDesktop)) && !slices.Contains(identity.Groups, string(types.RoleWindowsDesktop)) {
```

Comment:
```
	// SystemRoles is a list of system roles (e.g. auth, proxy, node, etc) used
	// in "multi-role" certificates. Single-role certificates encode the system role
	// in `Groups` for back-compat reasons.
```

Couldn't this check therefore mistakenly fail on some single-role certificates? cc @fspmarshall